crates/provider/src/kubernetes/namespace.rs

```diff
@@ -100,7 +107,21 @@ where
     }
 
     async fn initialize_k8s(&self) -> Result<(), ProviderError> {
-        let labels = BTreeMap::from([("foo".to_string(), "bar".to_string())]);
+        // TODO (javier): check with Hamid if we are using this labels in any scheduling logic.
+        let labels = BTreeMap::from([
+            (
+                "jobId".to_string(),
+                env::var("CI_JOB_ID").unwrap_or("".to_string()),
+            ),
+            (
+                "projectName".to_string(),
+                env::var("CI_PROJECT_NAME").unwrap_or("".to_string()),
+            ),
+            (
+                "projectId".to_string(),
+                env::var("CI_PROJECT_ID").unwrap_or("".to_string()),
+            ),
+        ]);
```

Comment:
These labels doesn't need to be namespaced ? (like kubernetes.io/app=somevalue ...)

Reply:
No, we don't need to use namespace here since those are only used internally.